### PR TITLE
Add CreateNonClassParent DTO

### PR DESCRIPTION
**Pull request by rmainseas**
This PR introduces the DTO representing the shape of the data being used to create a non-class parent on the server. Child data such as non-class events are not included in this DTO because those will be generated automatically by the server. Similarly meetings will be added on an as-needed basis by the user using the GUI, so those aren't sent over either.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:

- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [ ] I have added tests to cover my changes.

## Priority:
- [x] Normal 
- [ ] High 

## Related Issues:

Delivers the first piece of #389

### CODE REVIEW

File: src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    nullable: true,
+  })
+  public contactEmail?: string;
+
+  @ApiProperty({
+    description: 'The contact phone number for this non-class parent',
+    example: '(123) 456-7890',
+    nullable: true,
+  })
+  public contactPhone?: string;
```
Comment (jonseitz) [inline src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts:38]: As with email above, there is an `@IsPhoneNumber` validator that we can use with `@IsOptional`. It does take a country code as an argument, which we'd want to set to US (though I am _slightly_ concerned that we could have people wanting to use non-US phone numbers here, that seems like an edge case when we're managing physical spaces on a US campus.)

File: src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    nullable: true,
+  })
+  public notes?: string;
+
+  @ApiProperty({
+    description: 'Non class parent predicted enrollment size',
+    example: 82,
+    nullable: true,
+  })
+  public expectedSize?: number;
```
Comment (jonseitz) [inline src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts:52]: Should have `@IsInt`and `@IsOptional` validators

**Review by jonseitz — CHANGES_REQUESTED** [review]
Just a couple points below about adding [additional class-validator decorators][decorators] to some of these fields. Even if they're duplicating our type declarations, the validators will perform runtime validation and return richer server errors

[decorators]: https://github.com/typestack/class-validator#validation-decorators

**Review by jonseitz — CHANGES_REQUESTED** [review]
Just a couple points below about adding [additional class-validator decorators][decorators] to some of these fields. Even if they're duplicating our type declarations, the validators will perform runtime validation and return richer server errors

[decorators]: https://github.com/typestack/class-validator#validation-decorators

File: src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    description: 'Area primary key ID',
+  })
+  @IsUUID()
+  public area: string;
+
+  @ApiProperty({
+    description: 'The title of this non class parent',
+    example: 'Reading group',
+  })
+  public title: string;
```
[the remark this answers is not in the document]
Comment (rmainseas) [reply src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts:18]: You're absolutely right, it should. Fixed in 5ce1671152dbd7199aea9d5a6996bdfe6d161a6e

File: src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts (hunk 2 above)
Comment (rmainseas) [reply src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts:61]: Fixed in 5ce1671152dbd7199aea9d5a6996bdfe6d161a6e

File: src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts (hunk 1 above)
Comment (rmainseas) [reply src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts:44]: Yeah, I think that's somewhat valid but as you mention - it seems extremely unlikely that someone on campus booking a physical space would have a non-US phone number. If this were an app designed for international consumption, perhaps but I agree it seems like a very extreme edge case. If it does come up, we can always loosen the validation to take account of this...

Comment (rmainseas) [reply src/common/dto/nonClassMeetings/CreateNonClassParent.dto.ts:44]: Fixed in Fixed in 5ce1671152dbd7199aea9d5a6996bdfe6d161a6e

**Review by jonseitz — APPROVED** [review]
Approved

**Comment by rmainseas** [comment]
Thanks @jonseitz !

**Pull request merged**